Pocket/pocket-ios
# Widgets - Configurable Recommendations Widget

## Summary
* This PR updates the Recommendations widget so that users can select their favorite topic by long-pressing and selecting "Edit Widget"

## Implementation Details
* Moved Widgets code out of `PocketKit` and added it directly to `ItemWidgets` target: the new widget, based on `AppIntentConfiguration`, did not seem to work with the previous configuration. Investigation is still in progress and once (and if) the issue is resolved, it will be put back into `PocketKit`
* Add `TopicRecommendationsWidget`, an intent-based, configurable Recommendations widget
* Add `NewRecommendationsWidgetTip`, a tip displaying in Home that informs about the changes in the Recommendations widget

## Test Steps
* Build/run
* Go to Home and notice the tip (see animation below)
* Install a Recommendations widget, or go to an already installed Recommendations widget, if you were installing over an existing build
* Long press and make sure you see the "Edit Widget" menu item
* Tap on it, and choose a topic of your liking
* Make sure the topic displays in the widget
* Make sure the widget behaves as expected, redirecting to Pocket and opening an article when you tap on one of them.

## PR Checklist:

## Animation
<p align=center>
<img width=300 src=https://github.com/Pocket/pocket-ios/assets/34376330/1b1b85f9-3782-49f0-9154-b7b874d159d1>
</p>
